Classes/Controller/ManagementController.php

```diff
@@ -0,0 +1,331 @@
+        $this->view->setTemplate($templateName);
+        $this->view->setTemplateRootPaths(['EXT:qbank/Resources/Private/Templates/Management']);
+        $this->view->setPartialRootPaths(['EXT:qbank/Resources/Private/Partials']);
+        $this->view->setLayoutRootPaths(['EXT:qbank/Resources/Private/Layouts']);
+        $this->view->getRequest()->setControllerExtensionName('Qbank');
+        $this->view->assign(
+            'settings',
+            [
+                'dateFormat' => $GLOBALS['TYPO3_CONF_VARS']['SYS']['ddmmyy'],
+            ],
+        );
+        // Info window is included in this.
+        $this->moduleTemplate->getPageRenderer()->loadRequireJsModule('TYPO3/CMS/Filelist/FileList');
+    }
+
+    /**
+     * Generates the dropdown menu.
+     *
+     * @param ServerRequestInterface $request
+     * @param string $action
+     * @return void
+     */
+    private function generateDropdownMenu(ServerRequestInterface $request, string $action): void
+    {
+        $uriBuilder = GeneralUtility::makeInstance(UriBuilder::class);
+        $lang = $this->getLanguageService();
+        $lang->includeLLFile('EXT:qbank/Resources/Private/Language/locallang.xlf');
+        $menu = $this->moduleTemplate->getDocHeaderComponent()->getMenuRegistry()->makeMenu();
+        $menu->setIdentifier('WebFuncJumpMenu');
+
+        foreach ($this->actions as $menuAction) {
+            $menuItem = $menu
+                ->makeMenuItem()
+                ->setActive($action === $menuAction)
+                ->setHref(
+                    $uriBuilder->buildUriFromRoute('file_qbank', ['action' => $menuAction])
+                )
+                ->setTitle($lang->getLL('be.menu_item.' . $menuAction));
+            $menu->addMenuItem($menuItem);
+        }
+
+        $this->shortcutName = $lang->getLL('be.menu_item.qbank_overview');
+        $this->moduleTemplate->getDocHeaderComponent()->getMenuRegistry()->addMenu($menu);
+    }
+
+    /**
+     * Gets all buttons for the docHeader.
+     * @param mixed $action
+     */
+    private function generateButtons($action): void
+    {
+        $uriBuilder = GeneralUtility::makeInstance(UriBuilder::class);
+        $buttonBar = $this->moduleTemplate->getDocHeaderComponent()->getButtonBar();
+
+        if ($action === 'mappings') {
+            $newRecordButton = $buttonBar->makeLinkButton()
+                ->setHref((string)$uriBuilder->buildUriFromRoute(
+                    'record_edit',
+                    [
+                        'edit' => [
+                            'tx_qbank_domain_model_mapping' => ['new'],
+                        ],
+                        'returnUrl' => (string)$uriBuilder->buildUriFromRoute('file_qbank', ['action' => 'mappings']),
+                    ]
+                ))
+                ->setTitle($this->getLanguageService()->getLL('be.button.add_mapping'))
+                ->setIcon($this->iconFactory->getIcon('actions-add', Icon::SIZE_SMALL));
```

Comment:
Maybe good to show the label for UX reasons?

```suggestion
                ->setIcon($this->iconFactory->getIcon('actions-add', Icon::SIZE_SMALL))
                ->setShowLabelText(true);
```